### PR TITLE
pkg module documentation, and some additional cleanup

### PR DESCRIPTION
**Pull request by ameihm0912**
Added documentation, ParamsCreator(), and some cleanup.

### CODE REVIEW

File: src/mig/modules/pkg/doc.rst [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+in a compressed base64 encoded format. The agent then decodes this data, and
+pushes the OVAL definitions through the mozoval processor. The result of definitions
+are then returned to the investigator, indicating a true or false status.
+
+To use OVAL processing mode, the `oval` option is supplied on the command line,
+giving a properly formatted OVAL definition file as an argument.
+
+The `includefalse` parameter can be specified if false evaluations should be
+returned to the investigator. By default. only OVAL definitions that have
+evaluated to true are returned.
```
Comment (jvehent) [inline src/mig/modules/pkg/doc.rst]: This is a bit confusing. You may want to say that, by default, the module will only return packages that are deemed `vulnerable` based on the OVAL definitions, and by setting `includefalse`, the module will also return packages that are not vulnerable.

or something like that...

File: src/mig/modules/pkg/doc.rst [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+Currently the following package managers are supported for query on an agent system.
+
+* **RPM**: RPM based package managers, for Red Hat, CentOS, etc
+* **dpkg**: DPKG based package managers, for Debian, Ubuntu, etc
+
+Packages managed in other package managers will not be returned by the module.
+
+Notes on OVAL execution
+-----------------------
+Note that the mozoval library does not implement the full OVAL reference or
```
Comment (jvehent) [inline src/mig/modules/pkg/doc.rst]: maybe a link to said reference?

**Comment by jvehent** [comment]
looks good to me, if not for those 2 nits in the doc

**Comment by ameihm0912** [comment]
@jvehent Updated based on comments, feel free to review when you have a chance

**Pull request merged**